server/src/controllers/PartyController.ts

```diff
@@ -50,11 +53,38 @@ export class PartyController {
 
   public get = async (req: Request, res: Response) => {
     const address = req.params.address;
-    await this.getByAddress(address, res, party =>
-      PartyController.mapAndReturnPartyData(address, res, party)
+    await this.getByAddress(address, res, async party =>
+      PartyController.mapAndReturnPartyData(
+        address,
+        res,
+        party,
+        await this.getPartyData(party)
+      )
     );
   };
 
+  private getPartyData = async (party: IParty): Promise<any> => {
+    const grails = await this.grailCollection.find({
+      address: {
+        $in: party.userlist
+      }
+    });
```

Comment:
you should add a select here, to only return the partyData, this will - in theory-  make the query much faster :

```
const cursor = db
  .collection('inventory')
  .find({
    status: 'A'
  })
  .project({ item: 1, status: 1 });
```

Reply:
Yes, good point! Will update this